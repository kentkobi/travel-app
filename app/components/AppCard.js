import React from 'react';
import { View, Image, StyleSheet } from 'react-native';


import AppColors from '../config/AppColors';
import AppText from '../components/AppText';

function AppCard({category, title, subtitle, image}) {
    return (
        <View style={styles.container}>
            {isFinite(image)? <Image source={image} style={styles.image}/> :<Image source={{uri: image}} style={styles.image}/>}
            <AppText style={styles.title}> {title} </AppText>
            <AppText style={styles.subtitle}> {subtitle} </AppText>  
            <AppText style={styles.subtitle}> {category} </AppText> 
        </View>
    );
}



const styles = StyleSheet.create({
    container:{
        backgroundColor:AppColors.otherColorLite,
        borderRadius:20,
        overflow:"hidden",
        marginBottom: 25,
    }, 
    image:{
        height: 200,
        width: "100%",
    },
    title:{
        fontWeight:"bold",
    },
    subtitle:{
        fontSize:16,
    }
})

export default AppCard;