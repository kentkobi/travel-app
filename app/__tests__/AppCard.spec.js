import React from "react";
import renderer from "react-test-renderer";

import AppCard from "../components/AppCard";

describe("AppCard component", () => {
  test("renders correctly", () => {
    const item = {
      category: "category1",
      title: "title1",
      subtitle: "subtitle1",
      image: require("../__tests__/AppCardTestImage.jpeg")
    };

    const json = renderer
      .create(
        <AppCard
          title={item.title}
          subtitle={item.subtitle}
          image={item.image}
          category={item.category}
        />
      )
      .toJSON();
    expect(json).toMatchSnapshot();
  });

  test("contains image", () => {
    const item = {
      category: "category2",
      title: "title2",
      subtitle: "subtitle2",
      image: require("../__tests__/AppCardTestImage.jpeg")
    };

    const json = renderer
      .create(
        <AppCard
          title={item.title}
          subtitle={item.subtitle}
          image={item.image}
          category={item.category}
        />
      )
      .toJSON();

    expect(json.children.includes("AppCardTestImage.jpeg"));
  });

  test("contains title, subtitle, and category", () => {
    const item = {
      category: "category3",
      title: "title3",
      subtitle: "subtitle3",
      image: require("../__tests__/AppCardTestImage.jpeg")
    };

    const json = renderer
      .create(
        <AppCard
          title={item.title}
          subtitle={item.subtitle}
          category={item.category}
        />
      )
      .toJSON();

    expect(json.children.includes("title3"));
    expect(json.children.includes("subtitle3"));
    expect(json.children.includes("category3"));
  });
});
