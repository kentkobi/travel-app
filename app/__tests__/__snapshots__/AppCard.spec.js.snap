// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppCard component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#B8C5D6",
      "borderRadius": 20,
      "marginBottom": 25,
      "overflow": "hidden",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://reactnative.dev/docs/assets/p_cat2.png",
      }
    }
    style={
      Object {
        "height": 200,
        "width": "100%",
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Avenir-Roman",
          "fontSize": 20,
        },
        Object {
          "fontWeight": "bold",
        },
      ]
    }
  >
     
     
    title1
     
     
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Avenir-Roman",
          "fontSize": 20,
        },
        Object {
          "fontSize": 16,
        },
      ]
    }
  >
     
     
    subtitle1
     
     
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Avenir-Roman",
          "fontSize": 20,
        },
        Object {
          "fontSize": 16,
        },
      ]
    }
  >
     
     
    category1
     
     
  </Text>
</View>
`;
